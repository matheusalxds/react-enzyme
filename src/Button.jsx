import React from 'react';

export default props => (
  <button className={`${props.btnStyle}`} onClick={props.onClick} type={'button'}>
    {props.label}
  </button>
);

