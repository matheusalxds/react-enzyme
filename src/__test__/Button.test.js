import React from 'react';
import {shallow} from 'enzyme';
import Button from '../Button';
import sinon from 'sinon';

describe('Button', () => {
  it('should be defined', () => {
    expect(Button).toBeDefined();
  });

  it('should render correctly with name', () => {
    const wrapper = shallow(<Button label='button test'/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly but without name', () => {
    const wrapper = shallow(<Button label=''/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render static HTML', () => {
    const wrapper = shallow(<Button label='Label'/>);
    expect(wrapper.text()).toEqual('Label');
  });

  it('simulates onClick event', () => {
    const mockOnClick = sinon.spy();
    let wrapper = shallow(<Button onClick={mockOnClick}/>);
    wrapper.find('button').simulate('click');
    expect(mockOnClick.calledOnce).toEqual(true);
  });

  it('render style danger', () => {
    let wrapper = shallow(<Button btnStyle={'danger'}/>);
    expect(wrapper.find('button').hasClass('danger')).toEqual(true);
  });

  it('render style primary', () => {
    let wrapper = shallow(<Button btnStyle={'primary'}/>);
    expect(wrapper.find('button').hasClass('primary')).toEqual(true);
  });
});
