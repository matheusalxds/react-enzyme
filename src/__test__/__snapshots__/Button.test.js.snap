// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly but without name 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    label=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "",
      "className": "undefined",
      "onClick": undefined,
      "type": "button",
    },
    "ref": null,
    "rendered": "",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "",
        "className": "undefined",
        "onClick": undefined,
        "type": "button",
      },
      "ref": null,
      "rendered": "",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Button should render correctly with name 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    label="button test"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "button test",
      "className": "undefined",
      "onClick": undefined,
      "type": "button",
    },
    "ref": null,
    "rendered": "button test",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "button test",
        "className": "undefined",
        "onClick": undefined,
        "type": "button",
      },
      "ref": null,
      "rendered": "button test",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
